docs/testplan/Smartswitch-test-plan.md

```diff
@@ -0,0 +1,787 @@
+   
+#### Verify in
+ * Switch
+   
+#### Sample Output
+```
+On Switch:
+
+root@sonic:~#show reboot-cause
+
+Device          Name                    Cause                       Time                                User    Comment
+
+switch          2023_10_20_18_52_28     Watchdog:1 expired;         Wed 20 Oct 2023 06:52:28 PM UTC     N/A     N/A
+DPU3            2023_10_03_18_23_46     Watchdog: stage 1 expired;  Mon 03 Oct 2023 06:23:46 PM UTC     N/A     N/A
+DPU2            2023_10_02_17_20_46     reboot                      Sun 02 Oct 2023 05:20:46 PM UTC     admin   User issued 'reboot'
+
+root@sonic:~#show reboot-cause history
+
+Device          Name                    Cause                       Time                                User    Comment
+
+switch          2023_10_20_18_52_28     Watchdog:1 expired;         Wed 20 Oct 2023 06:52:28 PM UTC     N/A     N/A
+switch          2023_10_05_18_23_46     reboot                      Wed 05 Oct 2023 06:23:46 PM UTC     user    N/A
+DPU3            2023_10_03_18_23_46     Watchdog: stage 1 expired;  Mon 03 Oct 2023 06:23:46 PM UTC     N/A     N/A
+DPU3            2023_10_02_18_23_46     Host Power-cycle            Sun 02 Oct 2023 06:23:46 PM UTC     N/A     Host lost DPU
+DPU3            2023_10_02_17_23_46     Host Reset DPU              Sun 02 Oct 2023 05:23:46 PM UTC     N/A     N/A
+DPU2            2023_10_02_17_20_46     reboot                      Sun 02 Oct 2023 05:20:46 PM UTC     admin   User issued 'reboot'
+
+"show reboot-cause history module-name"
+
+root@sonic:~#show reboot-cause history dpu3
+
+Device      Name                    Cause                           Time                                User    Comment 
+   
+DPU3        2023_10_03_18_23_46     Watchdog: stage 1 expired;      Mon 03 Oct 2023 06:23:46 PM UTC     N/A     N/A
+DPU3        2023_10_02_18_23_46     Host Power-cycle                Sun 02 Oct 2023 06:23:46 PM UTC     N/A     Host lost DPU
+DPU3        2023_10_02_17_23_46     Host Reset DPU                  Sun 02 Oct 2023 05:23:46 PM UTC     N/A     N/A
+```
+
+#### Pass/Fail Criteria 
+ * Verify the output to check the latest reboot cause with the date time stamp at the start of reboot
+ * Verify all the reboot causes - Watchdog, reboot command, Host Reset
+
+
+### 1.12 Check the DPU state after OS reboot
+
+#### Steps
+
+Existing Test case:
+   * Reboot using a particular command (sonic reboot, watchdog reboot, etc) (timeout 5 mins, wait 2 mins)
+   * Wait for ssh to drop
+   * Wait for ssh to connect
+   * Database check –1 min timeout
+   * Check for uptime – (NTP sync)
+   * Check for critical process – 5 mins timeout – check every 20 secs
+   * Check for transceiver status –– 5 mins timeout – check every 20 seconds
+   * Check for pmon status
+   * Check for reboot cause 
+   * Reboot is successful
+   
+Reboot Test Case for DPU:
+ * After the exisiting case, Power on all the dpus using `config chassis modules startup <DPU_Number>`
```

Comment:
is there a reboot option to power on DPUs also automatically or do they need to be always started separately after the switch reboot is successful

Reply:
We dont have that reboot option, we have to power on dpus everytime